src/pages/home/home.ts

```diff
@@ -450,41 +523,48 @@ export class HomePage {
       .present();
   }
 
-  private selectLocation() {
-    if (this.selectedLocation && this.groupSocketService.userInfo.groupUID) {
-      this.groupSocketService.selectedLocation = {
-        socketId: '',
-        groupUID: this.groupSocketService.userInfo.groupUID,
-        latitude: this.selectedLocation.latitude,
-        longitude: this.selectedLocation.longitude,
-        title: this.selectedLocation.title,
-      };
-
-      this.groupSocketService.selectLocation();
-      this.navCtrl.pop();
-    }
+  private selectLocation(latitude, longitude, title) {
+    // alert('Inside selected location if statement');
+    this.groupSocketService.selectedLocation = {
+      socketId: '',
+      groupUID: this.groupSocketService.userInfo.groupUID,
+      latitude: latitude,
+      longitude: longitude,
+      title: title,
+    };
+
+    this.groupSocketService.selectLocation();
   }
 
   //If routed from the deeplink, join the room.
-  private joinHostGroup(currentPosition) {
-    if (this.host_uid && this.selectedLocation === undefined) {
-      this.groupSocketService.userInfo = {
-        socketID: '',
-        groupUID: this.host_uid,
-        username: this.username,
-        latitude: currentPosition.coords.latitude,
-        longitude: currentPosition.coords.longitude,
-      };
-
-      //Join the room specified by the group uid.
-      this.groupSocketService.joinGroup();
-      this.isInGroup = true;
-      this.navCtrl.pop();
-    }
+  private joinHostGroup(currentPosition, host_uid) {
+    //  alert(`Inside joinHostGroup function. uid is ${host_uid}`);
+    this.groupSocketService.userInfo = {
+      socketID: '',
+      groupUID: host_uid,
+      markerUID: this.userMarkerUID,
+      username: this.username,
+      latitude: currentPosition.coords.latitude,
+      longitude: currentPosition.coords.longitude,
+    };
+
+    //Join the room specified by the group uid.
+    this.groupSocketService.joinGroup();
+    this.isInGroup = true;
+  }
+
+  //Used to generate UIDs for handling marker removals.
+  private generateMarkerUID() {
```

Comment:
So this looks very weird here. This is definitely something that is already in some npm package somewhere.
I found one to generate uuid https://www.npmjs.com/package/uuid

Give it a shot!

Reply:
I was initially thinking of using a third party package, but I don't find it necessary when it's such a small fix. I checked out the package and it has a little too much overhead for what we need.